### PR TITLE
Rdf modules

### PR DESCRIPTION
**Pull request by FynnBe**
code structure changes as proposed in https://github.com/bioimage-io/spec-bioimage-io/pull/161#issuecomment-876567568

### CODE REVIEW

**Comment by FynnBe** [comment]
todo:
 - [ ] use delegated functions in tests

File: bioimageio/spec/delegator.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
```
Comment (constantinpape) [inline bioimageio/spec/delegator.py:1]: why is this file called delegator?

**Review by constantinpape — COMMENTED** [review]
The diff is a bit to large to properly review rn.
But I think it overall looks good; I find the name "delegator" a bit non-descript thoguh.
Also, are there any significant functions renamed or move such that importing them changes?

File: bioimageio/spec/delegator.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/spec/delegator.py:1]: it delegates the io function to the respective submodule

**Comment by FynnBe** [comment]
> Also, are there any significant functions renamed or move such that importing them changes?

everything importet from `bioimageio.spec` should work as before, but imports from previous submodules are broken.

File: bioimageio/spec/delegator.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/spec/delegator.py:1]: of course 'utils' or similar would also be appropriate, but there are already many files called 'utils.py' in here...

Comment (constantinpape) [reply bioimageio/spec/delegator.py:1]: Is this imported in the top-level, or do library users need to do `from spec.delegator import something`?

**Comment by constantinpape** [comment]
> everything importet from `bioimageio.spec` should work as before, but imports from previous submodules are broken.

Ok, I think that's good enough for now.

**Pull request merged**